README.md

```diff
@@ -7,59 +7,521 @@
-and parsed body must be passed to the plugin like so:
+Claim-based settlement is the simple case that this framework uses as its
+abstraction for settlement. Claim based settlement uses a unidirectional
+payment channel. You put your funds on hold, and give your peer signed claims
+for more and more of the funds. These signed claims are passed off-ledger, and
+your peer submits the highest claim when they want to get their funds.
+
+Claim based settlement has been implemented on ripple with the PayChan
+functionality, or on
+[bitcoin](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki) (and
+many other blockchains) by signing transactions that pay out of some script
+output.
 
 ```js
-plugin.receive(method, body)
+const { MakePaymentChannelPlugin } = require('ilp-plugin-virtual')
+const { NotAcceptedError } = require('ilp-plugin-shared').Errors
+const Network = require('some-example-network')
+const BigNumber = require('bignumber.js')
+
+return MakePaymentChannelPlugin({
+  // initialize fields and validate options in the constructor
+  constructor: function (ctx, opts) {
+    // we have one maxValueTracker to track the best incoming claim we've
+    // gotten so far. it starts with a value of '0', and contains no data.
+    ctx.state.bestClaim = ctx.backend.getMaxValueTracker('incoming_claim')
+
+    // the 'maxUnsecured' option is taken from the plugin's constructor.
+    // it defines how much the best incoming claim can differs from the amount
+    // of incoming transfers.
+    ctx.state.maxUnsecured = opts.maxUnsecured
+
+    // use some preconfigured secret for authentication
+    ctx.state.authToken = opts.authToken
+  },
+
+  // This token will be sent as a bearer token with outgoing requests, and used
+  // to authenticate the incoming requests. If the network you're using has a
+  // way to construct a shared secret, the authToken can be created
+  // automatically instead of being pre-shared and configured.
+  getAuthToken: (ctx) => (ctx.state.authToken),
+
+  // the connect function runs when the plugin is connected.
+  connect: async function (ctx) {
+    // network initiation should happen here. In a claim-based plugin, this
+    // would be the place to connect to the network and initiate payment
+    // channels if they don't exist already.
+    await Network.connectToNetwork()
+
+    // establish metadata during the connection phase
+    ctx.state.prefix = 'peer.network.' + (await Network.getChannelId())
+    ctx.state.account = await Network.getChannelSource()
+    ctx.state.peer = await Network.getChannelDestination()
+    ctx.state.info = {
+      prefix: ctx.state.prefix,
+      currencyScale: 6,
+      currencyCode: 'X??',
+      connectors: []
+    }
+  },
+
+  // Synchronous functions in order to get metadata. They won't be called until
+  // after the plugin is connected.
+  getAccount: (ctx) => (ctx.state.prefix + ctx.state.account),
```

Comment:
The plugin framework is making that guarantee that they don't have to work until the plugin is connected. I think throwing an error on `getAccount` is in the domain of boilerplate code handled by the framework.